lib/objects/active_time_ranges.dart

```diff
@@ -11,26 +11,33 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
+
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+part 'active_time_ranges.g.dart';
+
 /// The object holds active time range of a device.
 ///
-/// See document https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices#resource
-class ActiveTimeRanges {
-  String _date;
-  String _activeTime;
+/// See document https://developers.google.com/admin-sdk/directory/v1/reference/chromeosdevices#resource.
+
+abstract class ActiveTimeRanges
+    implements Built<ActiveTimeRanges, ActiveTimeRangesBuilder> {
+  /// Anonymus constructor
+  ActiveTimeRanges._();
 
-  /// Constructor with optional variables.
-  ActiveTimeRanges([this._date, this._activeTime]);
+  /// Factory constructor
+  static Serializer<ActiveTimeRanges> get serializer =>
+      _$activeTimeRangesSerializer;
 
-  /// Constructor form a json string.
-  ActiveTimeRanges.fromJson(Map<String, dynamic> json) {
-    _date = json['date'];
-    _activeTime = json['activeTime'];
-  }
+  /// Serializer to parse from Json
+  factory ActiveTimeRanges([void Function(ActiveTimeRangesBuilder) updates]) =
+      _$ActiveTimeRanges;
 
   /// Date of usage of the device.
-  String get date => _date;
+  @nullable
+  String get date;
 
   /// Duration of usage of the device in milliseconds.
-  String get activeTime => _activeTime;
+  @nullable
+  String get activeTime;
```

Comment:
You can do this instead on your classes, that puts in a toMap and a fromMap method on the built_value clases, then you don't need to call serializers.deserialize, you just call ActiveTimeRanges.fromMap()
Map<String, dynamic> toMap() {
    return serializers.serializeWith(Game.serializer, this);
  }

  static Game fromMap(Map<String, dynamic> jsonData) {
    return serializers.deserializeWith(Game.serializer, jsonData);
  }

  static Serializer<Game> get serializer => _$gameSerializer;

eg: https://github.com/pinkfish/flutter_basketball/blob/main/basketballdata/lib/data/game/game.dart

Reply:
Done, thanks for the advise it is easier to map objects from json and map objects to a json with this two functions.